src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,241 @@
+                if (!empty($releases)) {
+                    foreach ($releases as $release) {
+                        // Update to stable release, if available.
+                        if ($release == $stable_release) {
+                            $version = $release;
+                            break;
+                        }
+                    }
+                }
+            }
+        } else {
+            // Get the latest version from HEAD.
+            exec("cd \"$plugin\" && git rev-parse --abbrev-ref HEAD", $branch);
+            $version = !empty($branch) ? array_pop($branch) : 'unknown';
+        }
+        return $version;
+    }
+
+    /**
+     * Check whether a Packagist project is valid.
+     *
+     * @param string $project Packagist project name
+     * @return bool True if valid, false otherwise
+     */
+    protected function isValidPackagistProject($project)
+    {
+        // Search for the Packagist project.
+        exec("composer search -N -t terminus-plugin {$project}", $items);
+        if (!empty($items)) {
+            foreach ($items as $item) {
+                if ($item == $project) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check whether a URL is valid.
+     *
+     * @param string $url The URL to check
+     * @return bool True if the URL returns a 200 status, false otherwise
+     */
+    protected function isValidUrl($url = '')
+    {
+        // @TODO: This could be a generic utility function used by other commands.
+
+        if (!$url) {
+            return false;
+        }
+        $headers = @get_headers($url);
+        if (!isset($headers[0])) {
+            return false;
+        }
+        return (strpos($headers[0], '200') !== false);
+    }
+
+    /**
+     * Get the plugin Composer information.
+     *
+     * @param string $plugin Plugin name
+     * @return array of Composer information
+     */
+    protected function getComposerInfo($plugin)
+    {
+        // @TODO: This could be a generic utility function used by other commands.
+
+        $plugin_dir = $this->getPluginDir($plugin);
+        $composer_json = $plugin_dir . DIRECTORY_SEPARATOR . 'composer.json';
+        if (file_exists($composer_json)) {
+            $composer_data = @file_get_contents($composer_json);
+            return (array)json_decode($composer_data);
+        }
+        return array();
+    }
+
+    /**
+     * Platform independent check whether a command exists.
+     *
+     * @param string $command Command to check
+     * @return bool True if exists, false otherwise
+     */
+    protected function commandExists($command)
+    {
+        // @TODO: This could be a generic utility function used by other commands.
+
+        $windows = (php_uname('s') == 'Windows NT');
+        $test_command = $windows ? 'where' : 'command -v';
+        $file = popen("$test_command $command", 'r');
+        $result = fgets($file, 255);
```

Comment:
Would using Finder components help alleviate the need to check for system type at runtime? (Also helps w/ the testing)

Reply:
Sure, if you want to add some code snippet suggestion as a comment, I can add it or you can if you want permission to edit this branch.